### PR TITLE
[JENKINS-45845] Change containerCap and instanceCap 0 to mean do not use

### PR DESCRIPTION
**Pull request by carlossg**
With current behavior

It is inconsistent, probably needs to be fixed https://issues.jenkins-ci.org/browse/JENKINS-45845

### CODE REVIEW

**Comment by carlossg** [comment]
Seems that a instanceCap=0 should be valid, meaning "I don't want to use this template", it may require some changes 

ping @marvinthepa

**Comment by marvinthepa** [comment]
😃

**Comment by carlossg** [comment]
Added a possible implementation

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java [hunk 1]
```diff
@@ -216,7 +216,7 @@ public String getRemoteFs() {
     }
 
     public void setInstanceCap(int instanceCap) {
-        if (instanceCap <= 0) {
+        if (instanceCap < 0) {
```
Comment (marvinthepa) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java]: Maybe I am wrong, but doesn't this also need to go into [`readResolve`](https://wiki.jenkins.io/display/JENKINS/Hint+on+retaining+backward+compatibility) for backward compatibility?

Comment (carlossg) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java]: if you had instanceCap=0 the xml is written with `<instanceCap>2147483647</instanceCap>` so if I'm not mistaken there is nothing to do

Comment (marvinthepa) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java]: Sounds good.
I thought there might be a situation where the xml is loaded with `instanceCap` not set at all, but if that cannot occur I guess it's fine.

**Pull request merged**